packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -485,7 +485,10 @@ class FlutterPlugin implements Plugin<Project> {
             getPluginList().each { plugin ->
                 Project pluginProject = project.rootProject.findProject(plugin.key)
                 pluginProject.afterEvaluate {
-                    int pluginCompileSdkVersion = pluginProject.android.compileSdkVersion.substring(8) as int
+                    int pluginCompileSdkVersion = Integer.MIN_VALUE; // Default to int min if using a preview version to skip the sdk check.
```

Comment:
Nit: can you move these comments to the line above instead of after.

Reply:
Moved both these comments above, and did the same for the comments for the app case